### PR TITLE
SiStrip Gain2 PCL: Introduce new gain fitting procedure 

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The purpose of this PR is to:  
   * introduce different cluster charge fitting procedures: 
       * `legacy` = pure Landau PDF (default for PCL)
       * `DDRng` = pure Landau PDF with data-driven fit range 
       * `DDRngAllConv` = convolution of Landau with Gaussian PDF for all APVs
       * `DDRngConvExceptTOBL5L6` = convolution of Landau with Gaussian PDF for all APVs excepted for TOB L5 and L6;
   * add the possibility to retrieve an additional control ntuple out of the `SiStripGainsPCLHarvester` (by default off)
   * add calibtree -> `ALCAPROMPT` conversion plugin (`SiStripGainsCalibTreeWorker`) with example configs;
   * add unit tests for exercising the  calibtree -> `ALCAPROMPT` conversion plugin
   * https://github.com/cms-sw/cmssw/pull/36747/commits/a4dea878ea55c38390d9d4d1e2487ddeb7882ac9 gets rid of some static analyzer warnings within `CalibTracker/SiStripChannelGain` code. 
  
#### PR validation:

Relies on the unit tests which are introduced here: 

`scramv1 b runtests_testSSTGainPCL_fromCalibTree`

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
Note to reviewers: this PR requires https://github.com/cms-sw/cmssw/pull/36735 as a pre-requisite in order to pass unit tests.

**Comment by malbouis** [comment]
@mmusich , since this PR is modifying the SiStripsGain PCL harvester is there a particular workflow we should be testing this PR with?

**Comment by mmusich** [comment]
> since this PR is modifying the SiStripsGain PCL harvester is there a particular workflow we should be testing this PR with?

I tested it with 1001.0, which is included in the short matrix, and I don't think there are other wfs testing it. The package `CalibTracker/SiStripChannelGain` has unit tests though and those will be exercised as well. 
Finally please notice https://github.com/cms-sw/cmssw/pull/36747#issuecomment-1016492611

**Comment by malbouis** [comment]
test parameters:

- pull_request = #36735

**Comment by malbouis** [comment]
please test

**Comment by mmusich** [comment]
>-1
>
>Failed Tests: RelVals-INPUT
Summary: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b22f2/21823/summary.html
COMMIT: 6f0d819
CMSSW: CMSSW_12_3_X_2022-01-19-1100/slc7_amd64_gcc10
User test area: For local testing, you can use /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36747/21823/install.sh to create a dev area with all the needed externals and cmssw changes.

the issue reported in the test is not related to the PR and it seems is affecting all recent PRs. 
In the last push I am getting rid of some SA warnings in this package.

**Comment by malbouis** [comment]
please test

**Comment by malbouis** [comment]
please test

File: CalibTracker/SiStripChannelGain/plugins/SiStripGainFromData.cc [hunk 1]
```diff
@@ -888,8 +888,7 @@ void SiStripGainFromData::algoEndJob() {
 
     unsigned int GOOD = 0;
     unsigned int BAD = 0;
-    double MPVmean = MPVs->GetMean();
-    MPVmean = 300;
+    double MPVmean = 300.;  //MPVs->GetMean();
```
Comment (tvami) [inline CalibTracker/SiStripChannelGain/plugins/SiStripGainFromData.cc:891]: Should this really be hardcoded? Does having the `MPVs->GetMean()` alone lead to convergence issues?

Comment (mmusich) [reply CalibTracker/SiStripChannelGain/plugins/SiStripGainFromData.cc:891]: That's what was before. I am not changing the behavior, just removing the SA warning.

Comment (tvami) [reply CalibTracker/SiStripChannelGain/plugins/SiStripGainFromData.cc:891]: Right, I guess I could speak to the code authors, I was just surprised to see this magic number here

Comment (mmusich) [reply CalibTracker/SiStripChannelGain/plugins/SiStripGainFromData.cc:891]: that's not a magic number, it's the target MVP post-calibration.

**Comment by tvami** [comment]
+alca
 * Jenkins failure is unrelated
 * new unit test runs (although due to too low stats doesnt do much) https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3b22f2/21902/unitTests/src/CalibTracker/SiStripChannelGain/test/testSSTGainPCL_fromCalibTree/testing.log

**Comment by qliphy** [comment]
merge

**Pull request merged**